src/Storage/Repository/PgApplicationRepository.cs

```diff
@@ -0,0 +1,197 @@
+            }
+
+            return applications;
+        }
+
+        /// <inheritdoc/>
+        public async Task<Application> FindOne(string appId, string org)
+        {
+            if (!_memoryCache.TryGetValue(appId, out Application application))
+            {
+                await using NpgsqlCommand pgcom = _dataSource.CreateCommand(_readByIdSql);
+                pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, appId.Replace('/', '-'));
+                await using NpgsqlDataReader reader = await pgcom.ExecuteReaderAsync();
+                if (await reader.ReadAsync())
+                {
+                    application = SetLegacyId(reader.GetFieldValue<Application>("application"));
+                    _memoryCache.Set(appId, application, _cacheEntryOptionsMetadata);
+                }
+                else
+                {
+                    application = null;
+                }
+            }
+
+            return application;
+        }
+
+        /// <inheritdoc/>
+        public async Task<Application> Create(Application item)
+        {
+            SetInternalId(item);
+            await using NpgsqlCommand pgcom = _dataSource.CreateCommand(_createSql);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, item.Id);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, item.Org);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Jsonb, item);
+
+            await pgcom.ExecuteNonQueryAsync();
+
+            return SetLegacyId(item);
+        }
+
+        /// <inheritdoc/>
+        public async Task<Application> Update(Application item)
+        {
+            SetInternalId(item);
+            await using NpgsqlCommand pgcom = _dataSource.CreateCommand(_updateSql);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, item.Id);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Jsonb, item);
+            await pgcom.ExecuteNonQueryAsync();
+
+            SetLegacyId(item);
+            _memoryCache.Set(item.Id, item, _cacheEntryOptionsMetadata);
+            return item;
+        }
+
+        /// <inheritdoc/>
+        public async Task<bool> Delete(string appId, string org)
+        {
+            await using NpgsqlCommand pgcom = _dataSource.CreateCommand(_deleteSql);
+            pgcom.Parameters.AddWithValue(NpgsqlDbType.Text, appId.Replace('/', '-'));
+
+            return await pgcom.ExecuteNonQueryAsync() == 1;
+        }
+
+        /// <inheritdoc/>
+        public async Task<Dictionary<string, string>> GetAllAppTitles()
+        {
+            if (!_memoryCache.TryGetValue(_cacheKey, out Dictionary<string, string> appTitles))
+            {
+                appTitles = new Dictionary<string, string>();
+                foreach (Application item in await FindAll())
+                {
+                    StringBuilder titles = new();
+                    if (item?.Title?.Values != null)
```

Comment:
The "bug" being reported by SonarCloud might be linked to the question mark right after item. Try to remove it. It should be impossible for item to be null here. 
```suggestion
                    if (item.Title?.Values != null)
```

Reply:
Done